### PR TITLE
Issue #3: Setup new shopping list with autogenerated token

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can start tracking purchased items.
A shopping list consists of a set of items associated with a user’s token. Creating a new list consists of the following sequence:
      * Generate a new, unique token
      * Save the token to `localStorage`
      * Show the user the list view
      The script `getToken` from the `shopping-list-utils` package should be used to generate the token. More information on that package [here](https://www.npmjs.com/package/@the-collab-lab/shopping-list-utils).
      `import { getToken, words } from '@the-collab-lab/shopping-list-utils'`
      AC:
      * `@the-collab-lab/shopping-list-utils` has been added as a dependency to the project.
      * For users who do not already have a token or a list, a button or link exists on the home screen that allows them to create a new list
      * Clicking the button/link generates a new token and saves it to `localStorage`
      * Once the token has been created and saved, the user is redirected to the "list" view
      * For users who do have a token already saved in `localStorage`, they should be shown/redirected to the "list" view
      Notes:
      * Eventually the welcome/home page will look like the following wireframe: https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/welcome-screen.png
      * It's good for your whole team to look at the wireframe now, but be sure not to try to implement all of the visible functionality at once. There will be future stories to add the other features such as joining an existing list.
      * For this story, it's sufficient just to have a simple "create list" button. Although you are not required to save the token to Firestore, some teams have opted to also save the token to Firestore when generating a token.

### PR DESCRIPTION
**Pull request by djtaylor8**
## Description

- Imported getToken package to generate new tokens
- Added a button to create new list, get token, and redirect to List View
- Set new token in localStorage 
- Save token to Firestore for future use
- Use React hook (useEffect) to check for existing token in localStorage and redirect to List View

## Related Issue
Closes #3 

## Acceptance Criteria

* @the-collab-lab/shopping-list-utils has been added as a dependency to the project.
* For users who do not already have a token or a list, a button or link exists on the home screen that allows them to create a new list
* Clicking the button/link generates a new token and saves it to localStorage
* Once the token has been created and saved, the user is redirected to the "list" view
* For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
| ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

### After
-Added create list button in the Home component

Screen Shot 2022-01-17 at 2 28 41 PM

## Testing Steps / QA Criteria

-After pulling down our branch to your local environment, run ```npm install``` to get the latest dependencies. 
-When you click 'Create New List', the token will be saved in localStorage and you will automatically be redirected to the List View. To clear localStorage, go into your browser console and type ```localStorage.clear()```.

### CODE REVIEW

**Comment by rudidev08** [comment]
Awesome on including screenshots (it's good archive for UI state few months down the road when running old code isn't feasible), and testing steps. QA, other devs, and PMs would all find RP very helpful.

**Review by rudidev08 — APPROVED** [review]

File: src/components/Home.jsx [hunk 1]
```diff
@@ -1,10 +1,41 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { collection, addDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+
+import { getToken } from '@the-collab-lab/shopping-list-utils';
```
Comment (rudidev08) [inline src/components/Home.jsx:7]: Great job on being specific when doing imports!

**Review by alenamedved — APPROVED** [review]
LGTM. Great job.  Easy to follow steps, good PR.

File: src/components/Home.jsx [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
 
 function Home() {
+  const navigate = useNavigate();
```
Comment (lindseyindev) [inline src/components/Home.jsx:10]: I'm learning new things!!!

File: src/components/Home.jsx [hunk 3]
```diff
@@ -1,10 +1,41 @@
...
+    const token = getToken();
+    localStorage.setItem('list-token', token);
+
+    //add token to Firestore
+    try {
+      const docRef = await addDoc(collection(db, 'token'), {
+        token,
+      });
+      navigate('/listView');
+      console.log('Document written with ID: ', docRef.id);
```
Comment (lindseyindev) [inline src/components/Home.jsx:28]: is the docRef.id the token?

File: src/components/Home.jsx [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
+  const handleNewList = async () => {
+    const token = getToken();
+    localStorage.setItem('list-token', token);
+
+    //add token to Firestore
+    try {
+      const docRef = await addDoc(collection(db, 'token'), {
+        token,
+      });
+      navigate('/listView');
```
Comment (lindseyindev) [inline src/components/Home.jsx:27]: the navigation is SO user friendly to bypass the create list page if the user has a token in local storage and redirect them after they have created their list & token

**Review by lindseyindev — APPROVED** [review]

File: src/components/Home.jsx (hunk 2 above)
Comment (djtaylor8) [reply src/components/Home.jsx:10]: Woooo React Router v6!

File: src/components/Home.jsx (hunk 3 above)
Comment (djtaylor8) [reply src/components/Home.jsx:28]: No, this is the unique id assigned to the token by Firebase. This is kind of a default console.log, it's not super necessary but can be helpful if there's an error. The token itself is a 3 word string - you should be able to see a collection of tokens in our Firestore db.

**Review by EdwardHinkle — APPROVED** [review]
Nothing new to add beyond what everyone else has said :)

**Review by laurenyz — APPROVED** [review]

**Pull request merged**